### ISSUE TITLE
🚀 Feature: Add AWS Pinpoint as SMS provider

### ISSUE THREAD

**Issue by rdmchr** [novuhq/novu#1834]
### 🔖 Feature description

I saw that AWS SES is already supported as an Email provider, so I believe that it would be great if you could just use AWS. This would require Novu to support AWS Pinpoint for SMS delivery.

### 🎤 Why is this feature needed ?

As stated above if Novu supported AWS Pinpoint you could use AWS for everything novu related. (Hosting, sending email and sending SMS) I also saw that SNS is on the road map, so I think adding Pinpoint wouldn't be too much of a hassle.

### ✌️ How do you aim to achieve this?

I would consider contributing a PR. A lot of other providers are already implemented so I should be able to use them as a template.

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by scopsy** [comment]
@rdmchr sorry for the delayed response, but do you want to be assigned for this?

**Comment by rdmchr** [comment]
Yeah I would love to help out on this. But I don't have any experience when it comes to unit testing Pinpoint itself, so I might need some help there.

**Comment by scopsy** [comment]
Unfortunately I'm not very familiar with it either :( But maybe someone from our community is and will be able to help

**Comment by rdmchr** [comment]
Alright. Feel free to assign me and I'll figure something out ;)

**Comment by jainpawan21** [comment]
Hi @rdmchr Assigning to you, 
Ping us in discord or here if you need any help
